Update Dockerfile TZ env variable

ovargasp opened the pull request:
Makes Dockerfile use the TZ env variable from compose.yml file to set the timezone of the container, as it is it uses UTC even if that variable is set to a different timezone.

rocky4546 commented:
I see no issue with change. Two questions: 
First, does this impact when daylight saving time happens.  the purpose of having stuff on UTC is to allow for daylight savings time to adjust within each timezone independently.
Second, There are 3 Dockerfile files.  I thought the main one is [Dockerfile_tvh_crypt.alpine](https://github.com/cabernetwork/cabernet/blob/master/Dockerfile_tvh_crypt.alpine), but not a docker expert.  wondering if all 3 should see an update.

ovargasp commented:
For what I've read of the code https://github.com/cabernetwork/cabernet/blob/cf3dc4d8afb83912ba19385f618be02e06b8cb3f/lib/common/xmltv.py#L116 cabernet converts all dates to utc to create the epg in the xmltv.py, so it shouldn't affect daylight saving settings. Besides the only thing it does is enable the TZ variable in the container you can set the TZ value to UTC and it'll work the same. 

As for the Dockerfile, it depends on cabernet github's  configured workflow to create the docker image, https://github.com/cabernetwork/cabernet/blob/cf3dc4d8afb83912ba19385f618be02e06b8cb3f/.github/workflows/docker-image.yml#L1 it creates an image from master and dev branches using the Dockerfile definition, the other 2 Dockerfiles aren't refered by any other workflow nor the documentation so I think those were older/previous versions.

ovargasp commented:
I updated the pull request with another case that I found on processing Dates for EPG, using the same sourceas before: [epg.txt](https://github.com/user-attachments/files/18142362/epg.txt), like last time I found a date with a weird format: <date>2000|2001|2002|2003</date> it crashes the xmltv.py and prevents the rest of the epg data to be processed and aded to cabernet, so I added some code to catch the error and decided to use today's date as program date, it can be whatever you want it only prevents the crashing.

rocky4546 commented:
Need to incorporate this into the current set of changes.
